lib/rust/parser/src/macros.rs

```diff
@@ -0,0 +1,40 @@
+//! The macro system for the Enso parser.
+
+pub mod definition;
+pub mod literal;
+pub mod registry;
+
+use crate::prelude::*;
+use crate::prelude::logger::*;
+
+use crate::macros::definition::Definition;
+use crate::macros::registry::Registry;
+
+
+// ================
+// === Resolver ===
+// ================
+
+/// The Enso macro resolver.
+#[derive(Clone,Debug,PartialEq)]
+pub struct Resolver<Logger> {
+    /// The macro registry.
+    registry : Registry,
+    /// The logger for the macro resolver.
+    logger : Logger
+}
+
+impl<Logger:AnyLogger<Owned=Logger>> Resolver<Logger> {
```

Comment:
Move that to where clause in next line. We try to keep the things on the left minimal and easy to read